doc/source/whatsnew/v1.0.0.rst

```diff
@@ -39,8 +39,58 @@ Backwards incompatible API changes
 
 .. _whatsnew_1000.api.other:
 
-- :class:`pandas.core.groupby.GroupBy.transform` now raises on invalid operation names (:issue:`27489`).
--
+Groupby.transform(str) validates name is an aggregation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In previous releases, :meth:`DataFrameGroupBy.transform` and
+:meth:`SeriesGroupBy.transform`  did not validate that the function name
+passed was actually the name of an aggregation. As a result, users might get a
+cryptic error or worse, erroneous results. Starting with this release, these
+methods will rised if the name of a non-aggregation is passed to them. There
+is no change in the behavior associated with passing a callable.
+
+Users who relied on :meth:`DataFrameGroupBy.transform` or :meth:`SeriesGroupBy.transform`
+for transformations such as :meth:`DataFrameGroupBy.rank`, :meth:`DataFrameGroupBy.ffill`,
+etc, should instead call these method directly
+(:issue:`27597`) (:issue:`14274`) (:issue:`19354`) (:issue:`22509`).
+
+.. ipython:: python
+
+    df = pd.DataFrame([0, 1, 100, 99])
+    labels = [0, 0, 1, 1]
+    g = df.groupby(labels)
+
+*Previous behavior*:
+
+.. code-block:: ipython
+
+    In [1]: g.transform('ers >= Decepticons')
+    AttributeError: 'DataFrameGroupBy' object has no attribute 'ers >= Decepticons'
+
+    g.transform('rank')
+    Out[14]:
+         0
+    0  1.0
+    1  1.0
+    2  2.0
+    3  2.0
+
+    g.rank()
+    Out[15]:
+         0
+    0  1.0
+    1  2.0
+    2  2.0
+    3  1.0
+
+*New behavior*:
+
+.. ipython:: python
+   :okexcept:
+
+   g.transform('ers >= Decepticons')
```

Comment:
use foo, & make this a code-block (so we don't have the long traceback)

put the 'rank' in its own ipython block; I would also show .rank() or at least indicate that they are now the same.